projects/plugins/crm/modules/woo-sync/includes/class-woo-sync-background-sync-job.php

```diff
@@ -1213,6 +1214,15 @@ public function woocommerce_order_to_crm_objects(
 
 	    // ==== Tax Rates (on local stores only)
 	    if ( !$from_api ) {
+
+			// Force Woo order totals recalculation to ensure taxes were applied correctly
+			// Only force recalculation if the order is not paid yet
+			if ( $order->get_status() === 'pending' || $order->get_status() === 'on-hold' ) {
```

Comment:
Why not always?

Reply:
Because in WooCommerce, you only can recalculate orders while they are in one of those statuses, `pending` or `on-hold`. So we follow that behavior and don't auto-calculate orders in the other statuses.

Reply:
You're only thinking about the Woo order user interface. But in reality any Woo extension, API connection, etc., could modify the data in the order, so we should probably always recalculate.

Reply:
Believe me, I did it the first time, re-calculate it always, it was the first thing I did, but I got weird behavior (I cannot remember exactly), and that is why I had to make this condition.